utils/src/set_flag.rs

```diff
@@ -32,3 +32,9 @@ impl Deref for SetFlag {
         &self.0
     }
 }
+
+impl From<bool> for SetFlag {
+    fn from(value: bool) -> Self {
```

Comment:
This is problematic. There should be only one way to initialize this, then one way to set it, such that it's not used twice. This can ruin setting the value due to confusion.

In other words: This should always be initialized unset, then with a call becomes set.